unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/PlayerInfoCardHUD/PlayerInfoCardHUDController.cs

```diff
@@ -47,9 +51,34 @@ public void SetVisibility(bool visible)
         view.SetVisibility(visible);
     }
 
+    private void BlockPlayer()
+    {
+        if (!ownUserProfile.blocked.Contains(currentUserProfile.userId))
+        {
+            ownUserProfile.blocked.Add(currentUserProfile.userId);
+        }
+        view.Refresh();
+        WebInterface.SendBlockPlayer(currentUserProfile.userId);
+    }
+
+    private void UnblockPlayer()
+    {
+        if (ownUserProfile.blocked.Contains(currentUserProfile.userId))
+        {
+            ownUserProfile.blocked.Remove(currentUserProfile.userId);
+        }
+        view.Refresh();
+        WebInterface.SendUnlockPlayer(currentUserProfile.userId);
+    }
+
+    private void ReportPlayer()
+    {
+        WebInterface.SendReportPlayer(currentPlayerName);
+    }
+
     public void Dispose()
     {
-        if (currentUserProfile != null) 
+        if (currentUserProfile != null)
             currentUserProfile.OnUpdate -= SetUserProfile;
```

Comment:
We can use the `?` expression instead of `if` to make the code a bit more brief.
```suggestion
            currentUserProfile?.OnUpdate -= SetUserProfile;
```

Reply:
<img width="816" alt="Screen Shot 2020-02-11 at 17 08 22" src="https://user-images.githubusercontent.com/1307029/74274896-a195cd80-4cf1-11ea-90fd-c02761401942.png">